portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,6 +37,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
+  private Gson gson;
+
   private String convertToJsonUsingGson(ArrayList array) {
     Gson gson = new Gson();
```

Comment:
Sorry I wasn't super clear before. I also meant to put the the constructor call of Gson to an initialization. It seems to me for each Servlet, you could override the HttpServlet's init() method, like:

@Override
public void init(ServletConfig config) throws ServletException {
  super.init(config);
  // Do your own initialization here.
  gson = new Gson();
}

Then in your ConvertToJsonUsingGson() method, you could do:

private String convertToJsonUsingGson(ArrayList array) {
  return gson.toJson(array);
}